src/cooccur.c

```diff
@@ -263,7 +270,7 @@ int merge_files(int num) {
     for (i = 0; i < num; i++) {
         sprintf(filename,"%s_%04d.bin",file_head,i);
         fid[i] = fopen(filename,"rb");
-        if (fid[i] == NULL) {fprintf(stderr, "Unable to open file %s.\n",filename); return 1;}
+        if (fid[i] == NULL) {fprintf(stderr, "Unable to open file %s.\n",filename); free_fid(fid, num); free(pq); return 1;}
```

Comment:
FID was not zeroed out, so I believe this could lead to horrible failures when you try to close and free a non-existent file

Reply:
You are right, thank you. I updated free_fid(), it will close all previously opened files and skip NULLs.